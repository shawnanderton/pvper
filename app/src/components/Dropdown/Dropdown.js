import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import {
  FormControl,
  InputLabel,
  Select,
  MenuItem,
  ListItemIcon,
} from '@material-ui/core';

import Icon from '../Icon';
import './Dropdown.scss';

const useStyles = makeStyles((theme) => ({
  formControl: {
    margin: theme.spacing(1),
    minWidth: 160,
  },
  selectEmpty: {
    marginTop: theme.spacing(1),
  },
}));

function Dropdown({ onChange, label, items = [], selectedItems = [], imageType }) {
  const styles = useStyles();

  function handleChange(event) {
    const { name, value } = event.target;
    onChange({ name, items: value });
  }

  return (
    <FormControl className="dropdown">
      <InputLabel>{label}</InputLabel>
      <Select
        multiple
        name={label.toLowerCase()}
        value={selectedItems}
        onChange={handleChange}
        renderValue={(selected) => (
          <div className="label">
            {selected.map((c, i) => (
              <div key={i}>
                <Icon name={c} size={24} />
              </div>
            ))}
          </div>
        )}
      >
        {items.map((c, i) => (
          <MenuItem key={i} value={c}>
            <ListItemIcon>
              <Icon name={c} size={32} imageType={imageType} />
            </ListItemIcon>
            {c}
          </MenuItem>
        ))}
      </Select>
    </FormControl>
  );
}

Dropdown.propTypes = {
  onChange: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  items: PropTypes.array.isRequired,
  selectedItems: PropTypes.array,
};

export default Dropdown;
