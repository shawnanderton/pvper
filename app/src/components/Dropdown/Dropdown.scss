.dropdown {
    width: 90%;
    margin: 10px !important;
    .label {
        height: 60px;
        display: flex;
        flex-direction: column;
        flex-wrap: wrap;
        justify-content: center;
        align-items: center;
    }

}

