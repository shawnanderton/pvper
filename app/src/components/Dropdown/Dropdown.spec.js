import React from 'react';
import { shallow } from 'enzyme';
import Dropdown from './Dropdown';
import renderer from 'react-test-renderer';

describe('Dropdown', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Dropdown
          onChange={() => {}}
          label="hello"
          items={[]}
          selecedItems={[]}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the correct amount of menuItems', () => {
    const props = {
      label: 'Factions',
      items: ['allinance', 'horde'],
      onChange: () => {},
    };

    const wrapper = shallow(<Dropdown {...props} />);
    const menuItems = wrapper.find('WithStyles(ForwardRef(MenuItem))');
    expect(menuItems.length).toBe(2);
  });

  it('should render the correct amount of items selected', () => {
    const props = {
      label: 'Factions',
      items: ['allinance', 'horde'],
      selecteditems: ['allinance'],
      onChange: () => {},
    };

    const wrapper = shallow(<Dropdown {...props} />);
    const menuItems = wrapper.find('WithStyles(ForwardRef(Select))');
    expect(menuItems.length).toBe(1);
  });
});
