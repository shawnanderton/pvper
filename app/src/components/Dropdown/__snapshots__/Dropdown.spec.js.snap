// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render correctly 1`] = `
<div
  className="MuiFormControl-root dropdown"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
    data-shrink={false}
  >
    hello
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="mui-component-select-hello"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="mui-component-select-hello"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      name="hello"
      onAnimationStart={[Function]}
      type="hidden"
      value=""
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
